### PR TITLE
Fix #106 - Part 1: Add window_get_width().

### LINKED ISSUE #106: Simplify MenuState initialization
This will be an attempt to simplify a little bit the initialization of the `MenuState` in the `menu()` function. (I hope you are OK with this).

Substitute `menu_calculate_window_and_element_width(state)` with:

    int window_get_width()
    int menu_element_width(int n_columns, int page_size) // page_size == max_elements

Substitute `menu_calculate_rows_columns` with:

    // returns what is called "max_rows"
    int menu_rows_per_page(int line_height)

    // returns what is called "max_elements"
    int menu_page_size(int line_height, int num_elements)

    int menu_columns(int num_elements)

### PR DESCRIPTION
**Pull request by EdwinPujols**
This is the first part of #106. There are more to come!

### CODE REVIEW

**Comment by EdwinPujols** [comment]
*Note.* Some parts still look ugly and not DRY at all. But my intend is to ease the fix of #108 (natural width for hmode elements).

File: source/rofi.c [hunk 1]
```diff
@@ -951,78 +951,157 @@ static void calculate_window_position ( MenuState *state, const workarea *mon )
...
 }
 
+/**
+ * @param line_height The uniform height of a line of text.
+ *
+ * @returns Number of lines contained in the menu.
+ */
+static int menu_num_lines ( int line_height ) {
+    workarea mon;
+    monitor_active ( &mon );
```
Comment (DaveDavenport) [inline source/rofi.c]: 960 and 961 should be moved within the if statement (964) as they are only required in that scope.

File: source/rofi.c [hunk 2]
```diff
@@ -951,78 +951,157 @@ static void calculate_window_position ( MenuState *state, const workarea *mon )
...
+    int rpp = menu_num_lines ( line_height );
+
+    if ( config.fixed_num_lines == FALSE ) {
+        rpp = MIN ( rpp, (unsigned int) (
+                    ( num_elements + ( cols - num_elements % cols ) % cols ) / cols
+                    ) );
+    }
+
+    if ( config.hmode == TRUE ) {
+        rpp = 1;
```
Comment (DaveDavenport) [inline source/rofi.c]: this if block should be moved up to avoid unneeded calculations.

**Comment by EdwinPujols** [comment]
I am unsatisfied with the changes I've made. If you may, please close this issue, I'll try go slower and analyze the situation a bit deeper :)

**Pull request closed without merging**